### PR TITLE
Additions to testing & a fix

### PR DESCRIPTION
**Pull request by nuevoalex**
More tests and fixes the compilation/deployment issue where Maps are returning garbage.

### CODE REVIEW

File: source/tests-integration/TradeAndReport.ts [hunk 1]
```diff
@@ -11,21 +12,62 @@ describe("TradeAndReport", () => {
...
+        // Cancel the original rest of order
+        await fixture.cancelOrder(orderID);
+        const remainingAmount = await fixture.getOrderAmount(orderID);
+        expect(remainingAmount.toNumber()).to.equal(0);
+
+        // Proceed to reporting
+        let newTimestamp = await market.getEndTime_();
+        newTimestamp = newTimestamp.add(new BN(1));
+        await fixture.setTimestamp(newTimestamp);
+        let timestamp = await fixture.getTimestamp();
```
Comment (adrake33) [inline source/tests-integration/TradeAndReport.ts]: Should this be a constant?

File: source/tests-integration/TradeAndReport.ts [hunk 2]
```diff
@@ -11,21 +12,62 @@ describe("TradeAndReport", () => {
...
-        const outcome = new BN(0);
-        const numShares = new BN(10000000000000);
-        const price = new BN(2150);
+        const numTicks = await market.getNumTicks_();
+
+        // Place an order
+        let type = new BN(0); // BID
+        let outcome = new BN(0);
+        let numShares = new BN(10000000000000);
+        let price = new BN(2150);
```
Comment (adrake33) [inline source/tests-integration/TradeAndReport.ts]: Are these variables getting modified anywhere? Why are they no longer declared as constants?

File: source/tests-integration/TestFixture.ts [hunk 3]
```diff
@@ -92,6 +92,43 @@ export class TestFixture {
...
+
+        let actualPrice = price;
+        if (type == new BN(1)) {
+            const market = new Market(this.connector, this.accountManager, marketAddress, this.configuration.gasPrice);
+            const numTicks = await market.getNumTicks_();
+            actualPrice = numTicks.sub(price);
+        }
+        const ethValue = numShares.mul(actualPrice);
+
+        const bestPriceAmount = await trade.publicTakeBestOrder_(type, marketAddress, outcome, numShares, price, tradeGroupID, { attachedEth: ethValue });
```
Comment (adrake33) [inline source/tests-integration/TestFixture.ts]: Why does this function call both `publicTakeBestOrder_ ` and `publicTakeBestOrder`?

Comment (nuevoalex) [reply source/tests-integration/TestFixture.ts]: to get the return value. At present the EVm only support return values from eth calls, not transactions

File: source/tests-integration/TradeAndReport.ts (hunk 2 above)
Comment (nuevoalex) [reply source/tests-integration/TradeAndReport.ts]: At some point I intend to add more trading in here in which case they will be modified

**Pull request merged**